**Timothy Wen**

**Project 4**


The dataset that I am using can be found on https://data.montgomerycountymd.gov/Public-Safety/Crash-Reporting-Non-Motorists-Data/n7fk-dce5. The data surrounds the reports of car crashes with non-motorists such as pedestrians and cyclists. The dataset is provided by Montgomery County, MD. It provides information such as the substances drivers were on, weather conditions, light levels, and the severity of the crash on the pedestrians involved.

In [109]:

!wget https://data.montgomerycountymd.gov/api/views/n7fk-dce5/rows.csv


--2022-05-06 21:53:06--  https://data.montgomerycountymd.gov/api/views/n7fk-dce5/rows.csv
Resolving data.montgomerycountymd.gov (data.montgomerycountymd.gov)... 52.206.68.26, 52.206.140.199, 52.206.140.205
Connecting to data.montgomerycountymd.gov (data.montgomerycountymd.gov)|52.206.68.26|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/csv]
Saving to: ‘rows.csv.3’

rows.csv.3              [  <=>               ]   1.87M  6.53MB/s    in 0.3s    

2022-05-06 21:53:07 (6.53 MB/s) - ‘rows.csv.3’ saved [1960332]



In [110]:
import folium
import requests
import pandas as pd

# automatically downloads CSV file using the command above
crash_table = pd.read_csv('/content/rows.csv')

crash_table

Report Number  Local Case Number                Agency Name  \
0      MCP23480052          190024574   Montgomery County Police   
1      MCP20160048          190026280   Montgomery County Police   
2      MCP3171000V          190026764   Montgomery County Police   
3      MCP3000002N          190026219   Montgomery County Police   
4       EJ7876000N          190026354  Gaithersburg Police Depar   
...            ...                ...                        ...   
4533    EJ7876004K          220017728  Gaithersburg Police Depar   
4534   MCP3149003C          220017080   Montgomery County Police   
4535   MCP3084005S          220016909   Montgomery County Police   
4536   MCP209400JG          220012123   Montgomery County Police   
4537   MCP27230069          220016036   Montgomery County Police   

     ACRS Report Type         Crash Date/Time        Route Type  \
0        Injury Crash  05/23/2019 05:30:00 PM  Maryland (State)   
1        Injury Crash  06/02/2019 12:19:00 PM  Maryland (State)   
2        Injury Crash  06/04/2019 09:02:00 PM               NaN   
3        Injury Crash  06/01/2019 10:27:00 PM            County   
4        Injury Crash  06/02/2019 10:10:00 PM  Maryland (State)   
...               ...                     ...               ...   
4533     Injury Crash  04/26/2022 03:30:00 PM      Municipality   
4534     Injury Crash  04/22/2022 10:49:00 AM        US (State)   
4535     Injury Crash  04/21/2022 06:51:00 AM            County   
4536     Injury Crash  03/21/2022 11:05:00 AM            County   
4537     Injury Crash  04/15/2022 05:40:00 PM  Maryland (State)   

                   Road Name Cross-Street Type  \
0                GEORGIA AVE  Maryland (State)   
1             ROCKVILLE PIKE      Municipality   
2                        NaN               NaN   
3             MIDDLEBROOK RD            County   
4     MONTGOMERY VILLAGE AVE           Unknown   
...                      ...               ...   
4533           E DIAMOND AVE      Municipality   
4534           COLESVILLE RD              Ramp   
4535             CHESHIRE DR            County   
4536             QUINTANA DR            County   
4537             GEORGIA AVE            County   

                    Cross-Street Name  \
0                   UNIVERSITY BLVD W   
1                         ROLLINS AVE   
2                                 NaN   
3                   WARING STATION RD   
4               SPUR TO LOST KNIFE RD   
...                               ...   
4533                         ALLEY #3   
4534  RAMP 3 FR IS 495 EB TO US 29 NB   
4535                     GROSVENOR LA   
4536                        DWIGHT DR   
4537                         WENDY LA   

                                   Off-Road Description  ...  \
0                                                   NaN  ...   
1                                                   NaN  ...   
2     HOME DEPOT PARKING LOT\n 21010 FREDERICK RD GE...  ...   
3                                                   NaN  ...   
4                                                   NaN  ...   
...                                                 ...  ...   
4533                                                NaN  ...   
4534                                                NaN  ...   
4535                                                NaN  ...   
4536                                                NaN  ...   
4537                                                NaN  ...   

                 Pedestrian Actions               Pedestrian Location  \
0               NO IMPROPER ACTIONS           ON ROADWAY AT CROSSWALK   
1               NO IMPROPER ACTIONS                          SIDEWALK   
2               NO IMPROPER ACTIONS  AT INTERSECTION BUT NO CROSSWALK   
3                             OTHER  AT INTERSECTION MARKED CROSSWALK   
4                             OTHER           ON ROADWAY AT CROSSWALK   
...                             ...                               ...   
4533     

In [111]:
crash_table[pd.notnull(crash_table['Latitude'])]['Longitude'].count()

4538

We can see that the table for this has 4538 rows and all rows have a latitude and longitude coordinate the coorelates to a vehicle crash with a non-motorist location

**Impact of Clothing Color on Severity of Crash**

Description of dataset:

In this part, I analyze the rates of dark and light clothing for crash reports that result in serious injuries or fatal injuries for pedestrians. My belief is that a person wearing darker colors will have higher rates of severe crashes than pedestrians in light clothing because darker colors tend to blend in with the surrounding objects/landscape. Vivid colors such as yellow or orange may be more noticeable than black or brown.

In [112]:

map_osm = folium.Map(location=[39.15, -77.14], zoom_start=11)

for index, crash in crash_table[crash_table['Pedestrian Type'] == 'PEDESTRIAN'].iterrows():
    if crash['Injury Severity'] == 'SUSPECTED SERIOUS INJURY' or crash['Injury Severity'] == 'FATAL INJURY':
      if crash['Pedestrian Visibility'] == 'DARK CLOTHING':
        folium.Marker(location=[crash["Latitude"], crash["Longitude"]],icon=folium.Icon(color='black', icon="empty")).add_to(map_osm)
      if crash['Pedestrian Visibility'] == 'LIGHT CLOTHING':
        folium.Marker(location=[crash["Latitude"], crash["Longitude"]],icon=folium.Icon(color='white', icon="user")).add_to(map_osm)


map_osm

Description of Interactive Map:

Through a preliminary analysis of the map, we can observe that there does appear to be a greater amount of severe crashes involving pedestrians wearing darker clothes compared to those wearing lighter colors. Additionally, an unintended observation that can be made is that a majority of the severe/deadly crashes are located along highways, busy intersections/roads, and/or routes. One good example of this would be along Georgia Ave. near Wheaton where we can observe a cluster of non-motorist crashes along several major freeways.

This data could be effectively utilized by officials to increase visibility and lighting in clustered areas. Additionally, another solution could be painting colors that would contrast with darker clothes to help drivers be more careful.